cloud/pkg/controller/controller.go

```diff
@@ -32,30 +34,33 @@ func (ctl *Controller) Group() string {
 // Start controller
 func (ctl *Controller) Start(c *bcontext.Context) {
 	config.Context = c
+	ctl.stopChan = make(chan bool)
 	upstream, err := controller.NewUpstreamController()
 	if err != nil {
 		log.LOGGER.Errorf("new upstream controller failed with error: %s", err)
 		os.Exit(1)
 	}
 	upstream.Start()
 
-	for {
-		stopChannel := make(<-chan struct{})
-		downstream, err := controller.NewDownstreamController()
-		if err != nil {
-			log.LOGGER.Warnf("new downstream controller failed with error: %s", err)
-			continue
-		}
-		downstream.Start()
-
-		<-stopChannel
-		log.LOGGER.Warnf("elected slave, stopping downstream controller...")
-		downstream.Stop()
-		log.LOGGER.Warnf("downstream controller stopped")
+	downstream, err := controller.NewDownstreamController()
+	if err != nil {
+		log.LOGGER.Warnf("new downstream controller failed with error: %s", err)
+		os.Exit(1)
 	}
+	downstream.Start()
+
+	<-ctl.stopChan
+	log.LOGGER.Warnf("stopping upstream controller...")
+	upstream.Stop()
+	log.LOGGER.Warnf("upstream controller stopped")
+	log.LOGGER.Warnf("elected slave, stopping downstream controller...")
```

Comment:
I don't think we have any leader election mechanism , probably we can delete this log.

Reply:
Agree, Done, ptal. We can add the leader election in the future.